pre_commit/languages/rust.py

```diff
@@ -90,17 +173,23 @@ def install_environment(
             else:
                 packages_to_install.add((package,))
 
-        for args in packages_to_install:
-            cmd_output_b(
-                'cargo', 'install', '--bins', '--root', directory, *args,
-                cwd=prefix.prefix_dir,
-            )
+        with in_env(prefix, language_version):
+            toolchain = os.getenv('RUSTUP_TOOLCHAIN')
+            if toolchain is not None:
+                install_rust_with_toolchain(toolchain)
```

Comment:
rather than ping-ponging through environment variables as globals this should just check directly -- I believe that'd be `language_version != 'system'`

Reply:
We still need the toolchain type, but you're right that we don't need to get it from the env. See https://github.com/pre-commit/pre-commit/pull/2534/commits/7c22138df19335864f08738e2d860686c61194fa.